### PR TITLE
DCOS-38944 & DCOS-39074: add styling and dropdown truncation to header bar

### PR DESCRIPTION
**Pull request by tnguyen0**
* Adds the styling for the header bar elements: logo, side bar toggle, dropdowns
* Dropdowns are truncated and show ellipsis when the viewport is small
* Removes header bar from ui-kit-stage
* Enable the header bar

Closes DCOS-38944 & DCOS-39074

## Testing
* Checkout the branch on private referring to DCOS-38944
* Resize the screen and verify the dropdowns truncate
* The sidebar header is removed

## Result (For Truncation)
header-bar-recording

## Trade-offs
* Keep the full logo as long as possible
* When the max-width is too small to show any meaningful text on the dropdown (chosen to be 100px), collapse the logo and expand the size of the dropdown
* Small intervals so there is not too much space between the logo and dropdowns

## Dependencies
Hold merge until the following are merged:
~~DCOS-39457 (https://github.com/dcos/dcos-ui/pull/3086)~~
~~DCOS-38942~~
~DCOS-39679~

### CODE REVIEW

**Comment by tnguyen0** [comment]
@leemunroe can you have a look?

**Comment by leemunroe** [comment]
Gif looks good

File: packages/ui-kit-stage/HeaderBar.tsx [hunk 1]
```diff
@@ -4,21 +4,10 @@
  * JIRA: https://jira.mesosphere.com/browse/DCOS-39074
```
Comment (juliangieseke) [inline packages/ui-kit-stage/HeaderBar.tsx]: I think it is okay if you do DCOS-39074 in this PR 👍

File: src/js/components/AccountDropdownTrigger.js [hunk 2]
```diff
@@ -27,9 +27,11 @@ class AccountDropdownTrigger extends mixin(StoreMixin) {
 
     // TODO: DCOS-38944
```
Comment (juliangieseke) [inline src/js/components/AccountDropdownTrigger.js]: This PR is 38944 :)

File: src/js/components/HeaderBar.tsx [hunk 3]
```diff
@@ -5,19 +5,18 @@ import ClusterDropdown from "./ClusterDropdown";
 import AccountDropdown from "./AccountDropdown";
 
 export default function HeaderBar() {
-  // remove this to activate component
-  if (arguments) {
```
Comment (juliangieseke) [inline src/js/components/HeaderBar.tsx]: Logo Link & Sidebar are still missing - as far as I understood we can not enable this before these two things are done :(

File: src/styles/components/header-bar/styles.less [hunk 4]
```diff
@@ -65,9 +57,81 @@
...
+
+        &:after {
+          .header-bar-caret(@purple-lighten-4);
+        }
+
+        .header-bar-dropdown-content {
+
+          & when (@layout-screen-small-enabled) {
+
+            @media (max-width: @layout-screen-iteration-10) {
```
Comment (juliangieseke) [inline src/styles/components/header-bar/styles.less]: Uff… I'm not that familiar with css nowadays, but isnt there a better solution to this? 😢

File: packages/ui-kit-stage/HeaderBar.tsx [hunk 5]
```diff
@@ -4,21 +4,10 @@
...
-  flexShrink: 1,
-  flexGrow: 0,
-  flexBasis: "auto",
-  zIndex: 999
-};
-
 export class HeaderBar extends React.PureComponent {
   render() {
-    return <div style={headerBarStyles}>{this.props.children}</div>;
+    return <div className="header-bar">{this.props.children}</div>;
```
Comment (DanielMSchmidt) [inline packages/ui-kit-stage/HeaderBar.tsx]: Thanks for moving it into a class 👍

File: src/js/components/AccountDropdownTrigger.js [hunk 6]
```diff
@@ -27,9 +27,11 @@ class AccountDropdownTrigger extends mixin(StoreMixin) {
...
     return (
-      <a className="header-bar-dropdown" onClick={onTrigger}>
-        <span>{content}</span>
-      </a>
+      <span className="header-bar-dropdown-trigger" onClick={onTrigger}>
+        <span className="header-bar-dropdown-content text-overflow">
```
Comment (DanielMSchmidt) [inline src/js/components/AccountDropdownTrigger.js]: just a naming thing, but wouldn

File: src/js/components/HeaderBar.tsx [hunk 7]
```diff
@@ -1,24 +1,24 @@
 import * as React from "react";
 
-import { HeaderBar as UIHeaderBar } from "ui-kit-stage/HeaderBar";
 import ClusterDropdown from "./ClusterDropdown";
 import AccountDropdown from "./AccountDropdown";
 import { Link } from "react-router";
 
-export default function HeaderBar() {
-  // remove this to activate component
-  if (arguments) {
```
Comment (DanielMSchmidt) [inline src/js/components/HeaderBar.tsx]: Do we really want to enable this right now?

Comment (tnguyen0) [reply src/js/components/HeaderBar.tsx]: Yeah, this will be PR that enables it after its dependencies are merged.

**Comment by tnguyen0** [comment]
@leemunroe can you pull the branch and see how it looks

**Comment by leemunroe** [comment]
@tnguyen0 Nice the toggle interaction is looking great.

One thing that looks off is alignment. The goal was to have the hamburger icon aligned with the other icons in the sidebar. Right now that is not the case. Is this being dealt with in the "polish" task?

Goal
image
image

Current
image

**Comment by tnguyen0** [comment]
@juliangieseke @Fabs @DanielMSchmidt @weblancaster @nLight ready for a look 🙏

File: src/js/components/HeaderBar.tsx [hunk 8]
```diff
@@ -1,26 +1,26 @@
...
 import ClusterDropdown from "./ClusterDropdown";
 import AccountDropdown from "./AccountDropdown";
 import SidebarToggle from "./SidebarToggle";
 
-export default function HeaderBar() {
-  // remove this to activate component
-  // when enabled, remove skip from Sidebar-cy.js
-  if (arguments) {
-    return null;
+export class HeaderBar extends React.Component {
```
Comment (nLight) [inline src/js/components/HeaderBar.tsx]: wonder why default export didn't work?

File: plugins/oauth/hooks.js [hunk 9]
```diff
@@ -14,7 +13,6 @@ import CookieUtils from "#SRC/js/utils/CookieUtils";
 import RouterUtil from "#SRC/js/utils/RouterUtil";
 import UsersPage from "#SRC/js/pages/system/UsersPage";
 
-import AuthenticatedUserAccountDropdown from "./components/AuthenticatedUserAccountDropdown";
```
Comment (nLight) [inline plugins/oauth/hooks.js]: So we're removing this to add this back in a different PR?

File: src/js/pages/Index.js [hunk 10]
```diff
@@ -13,7 +13,7 @@ import MesosStateStore from "../stores/MesosStateStore";
 import Modals from "../components/Modals";
 import RequestErrorMsg from "../components/RequestErrorMsg";
 import ServerErrorModal from "../components/ServerErrorModal";
-import HeaderBar from "../components/HeaderBar";
+import { HeaderBar } from "../components/HeaderBar";
```
Comment (nLight) [inline src/js/pages/Index.js]: exporting by default would eliminate the need, wouldn't it?

**Comment by leemunroe** [comment]
This looks great! Thanks for addressing visual issues.

image

**Comment by nLight** [comment]
I don't see the second dropdown. Is this expected?

screen shot 2018-08-01 at 10 43 35

**Comment by tnguyen0** [comment]
@nLight can you get the latest of the corresponding privates plugin branch

File: plugins/oauth/components/AuthenticatedUserAccountDropdown.js [hunk 11]
```diff
@@ -5,46 +5,21 @@ import UserAccountDropdown from "#SRC/js/components/UserAccountDropdown";
...
-      userLabel = user.description;
-    } else if (user && user.is_remote) {
-      userLabel = user.uid;
-    }
+    this.userLabel = AuthStore.getUserLabel();
+  }
 
-    return userLabel;
+  getTrigger() {
+    return <UserAccountDropdownTrigger content={this.userLabel} />;
```
Comment (nLight) [inline plugins/oauth/components/AuthenticatedUserAccountDropdown.js]: we could just use `AuthStore.getUserLabel()` and get rid of constructor.

File: src/js/components/AccountHeader.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+      />
+    );
+  }
+}
+
+AccountHeader.contextTypes = {
+  router: routerShape
+};
+
+module.exports = AccountHeader;
```
Comment (nLight) [inline src/js/components/AccountHeader.js]: Let's go with `export default` instead

File: src/js/components/AccountHeader.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
+import { MountService } from "foundation-ui";
+import React from "react";
+import { routerShape } from "react-router";
+
+class AccountHeader extends React.Component {
```
Comment (nLight) [inline src/js/components/AccountHeader.js]: Any particular reason not doing a functional component?

File: src/js/components/UserAccountDropdown.js [hunk 14]
```diff
@@ -1,16 +1,14 @@
...
+import { Dropdown } from "reactjs-components";
 
 class UserAccountDropdown extends React.Component {
-  getMenuItems() {
-    return this.props.menuItems;
-  }
-
   getTrigger() {
-    const { clusterName } = this.props;
+    return this.props.trigger;
```
Comment (nLight) [inline src/js/components/UserAccountDropdown.js]: Let's remove the function and use the prop directly instead

File: src/js/components/UserAccountDropdown.js [hunk 15]
```diff
@@ -1,16 +1,14 @@
...
+  }
 
-    return <UserAccountDropdownTrigger primaryContent={clusterName} />;
+  getMenuItems() {
+    return this.props.menuItems;
```
Comment (nLight) [inline src/js/components/UserAccountDropdown.js]: same

File: src/js/components/UserAccountDropdownTrigger.js [hunk 16]
```diff
@@ -23,48 +23,22 @@ class UserAccountDropdownTrigger extends mixin(StoreMixin) {
...
-  showCaret: false,
-  onTrigger() {}
-};
-
 UserAccountDropdownTrigger.propTypes = {
-  clusterName: PropTypes.node,
+  content: PropTypes.string.isRequired,
   onUpdate: PropTypes.func,
-  showCaret: PropTypes.bool
+  onTrigger: PropTypes.func
```
Comment (nLight) [inline src/js/components/UserAccountDropdownTrigger.js]: usually we go with either `isRequired` or a default value for functions.

File: plugins/oauth/components/AuthenticatedUserAccountDropdown.js (hunk 11 above)
Comment (tnguyen0) [reply plugins/oauth/components/AuthenticatedUserAccountDropdown.js]: Then, I'd have to call `AuthStore.getUserLabel()` twice. Is that preferable to having a constructor?

Comment (nLight) [reply plugins/oauth/components/AuthenticatedUserAccountDropdown.js]: We used to call the logic twice, no strong opinion on that. Please use your best judgement

Comment (tnguyen0) [reply plugins/oauth/components/AuthenticatedUserAccountDropdown.js]: In that case, I will leave as is.

File: src/js/components/UserAccountDropdownTrigger.js (hunk 16 above)
Comment (tnguyen0) [reply src/js/components/UserAccountDropdownTrigger.js]: For this prop, it is not required. I don't see a default value to be set here.

**Comment by tnguyen0** [comment]
@nLight comments addressed

**Comment by DanielMSchmidt** [comment]
screen recording 2018-08-01 at 02 16 pm

Do we want to file this as a visual bug?

File: src/styles/components/header-bar/styles.less [hunk 17]
```diff
@@ -1,59 +1,72 @@
...
+    flex-basis: auto;
+    flex-grow: 0;
+    flex-shrink: 1;
+    height: @header-bar-height;
+    padding-left: @header-bar-horizontal-padding;
+    padding-right: @header-bar-horizontal-padding;
+    width: 100%;
+    z-index: 999;
+
+    div,
```
Comment (weblancaster) [inline src/styles/components/header-bar/styles.less]: let's use classes instead type selector.

**Comment by DanielMSchmidt** [comment]
@tnguyen0 This has merge conflicts

File: src/js/components/UserAccountDropdownTrigger.js (hunk 16 above)
Comment (DanielMSchmidt) [reply src/js/components/UserAccountDropdownTrigger.js]: I disagree, a trigger without such a function wouldn't make any sense and would be conceptually probably a misused, still I think this is not so important right now

Comment (weblancaster) [reply src/js/components/UserAccountDropdownTrigger.js]: I agree with @DanielMSchmidt here but not a blocker

**Pull request merged**